test/unit/view-layout/test-indexlayout.cpp

```diff
@@ -0,0 +1,296 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <gtest/gtest.h>
+#include "RAJA/util/types.hpp"
+#include "RAJA_test-base.hpp"
+
+using namespace RAJA;
+
+TEST(IndexLayout, IndexList1D) {
+  /*
+   * Construct a 1D index layout with the index list {1,2,3}
+   */
+
+  Index_type arr[3] = {1,2,3};
+
+  auto index_tuple = make_index_tuple(IndexList<>(&arr[0]));
+  auto index_layout = make_index_layout(index_tuple, 3);
+
+  EXPECT_EQ(index_layout(0), 1);
+  EXPECT_EQ(index_layout(1), 2);
+  EXPECT_EQ(index_layout(2), 3);
+
+}
+
+TEST(IndexLayout, IndexList1DSubsetOfLayout) {
+  /*
+   * Construct a 1D index layout of arbitrary size greater than 3 
+   * with the index list {2,3,4}
+   */
+
+  Index_type arr[3] = {2,3,4};
+
+  auto index_tuple = make_index_tuple(IndexList<>(&arr[0]));
+  auto index_layout = make_index_layout(index_tuple, 5);
```

Comment:
is this test testing to make sure you can have a larger stride than the length of the underlying IndexList?

Reply:
Yes, exactly.  This seems to be a common use case.  A user may want to create a view with some number of entries and index a subset of those.

Reply:
Does it actually test anything different in practice than the above test?

Reply:
My primary concern is people being confused about why its making a layout that would be easy to misuse. Perhaps the comment could be updated to say what you said in your response to my initial comment.

Reply:
I added some clarification to this unit test.  Let me know if you have any other comments on this, otherwise I'll resolve this thread.